### ISSUE TITLE
Combiner does not handle /web/ folder

### ISSUE THREAD

**Issue by aschempp** [contao/core-bundle#328]
The `Combiner` class always assumes scripts/stylesheets are in `TL_ROOT`. Same problem as we fixed in the image library.

**Comment by leofeyer** [comment]
The combiner actually works with absolute paths. Where exactly does the problem occur?

**Comment by aschempp** [comment]
That's exactly the problem. Combiner expects files to be in TL_ROOT but scripts are in /web/bundles/…

**Comment by leofeyer** [comment]
How do you add them to the combiner?

**Comment by aschempp** [comment]
$GLOBALS['TL_JAVASCRIPT'][] = "bundles/app/app.js|static";

**Comment by leofeyer** [comment]
See, that is wrong. You have to provide the correct path:

```php
$GLOBALS['TL_JAVASCRIPT'][] = "web/bundles/app/app.js|static";
```

**Comment by aschempp** [comment]
I disagree. In the backend, where there is no combiner, I would use the same variable but a relative root path.

**Comment by leofeyer** [comment]
But in the back end, you are specifying a **relative** path to be added to the document root. In the template however, you are specifying an **absolute** path.

**Comment by aschempp** [comment]
I always do specify a relative path?

**Comment by leofeyer** [comment]
Yes, but one is to be added to the base URL and one to the root path.

**Comment by aschempp** [comment]
Which was the same in Contao 3. And therefore is not BC.

**Comment by leofeyer** [comment]
Ok, that's a valid point.

**Comment by aschempp** [comment]
The fix should be fairly simple, same as for Image. Simply test both paths.

**Comment by aschempp** [comment]
On the other hand, scripts must always be in the web folder…

**Comment by leofeyer** [comment]
Actually, the combiner does not care at all. It works with absolute file paths and not with relative URL paths. You might as well add files from anywhere.

**Comment by aschempp** [comment]
So you mean the check should be in the PageRegular (or wherever the TL_JAVASCRIPT is combined) ?

**Comment by leofeyer** [comment]
I actually mean that you should specify the correct path (including `/web`).

**Comment by jamesdevine** [comment]
I'm having trouble with this too, maybe I'm misunderstanding how I should be using it but if I add a file to the global CSS array like this:

````
$GLOBALS['TL_CSS'][] = '/web/bundles/mybundle/mystyles.css||static';
````

It gets output into the template like this:

````
<link rel="stylesheet" href="/web/bundles/mybundle/mystyles.css">
````

**Comment by jamesdevine** [comment]
Just to clarify - i'm taking about when Contao is in debug mode and the scripts aren't being combined. When in normal mode and the scripts are being combined it works fine.

**Comment by leofeyer** [comment]
You should add it without the leading slash:

```php
$GLOBALS['TL_CSS'][] = 'web/bundles/mybundle/mystyles.css||static';
```

**Comment by leofeyer** [comment]
Fixed in 315dec7f32717c2f2226b0762ea8406b5ea80661.
